production_entry_app/production_entry_app/joint_production.py

```
@@ -0,0 +1,572 @@
+) -> dict[str, float]:
+	lh_weight = flt(lh_gross_qty) * flt(lh_bom_unit_cost)
+	rh_weight = flt(rh_gross_qty) * flt(rh_bom_unit_cost)
+	total_weight = lh_weight + rh_weight
+	if total_weight <= 0:
+		frappe.throw(_("Joint output BOM cost weight must be greater than zero."))
+	value = flt(net_production_value)
+	lh_value = value * lh_weight / total_weight
+	return {"LH": lh_value, "RH": value - lh_value}
+
+
+def materialize_joint_production_rows(doc: Any) -> list[dict[str, Any]]:
```

Ruff (0.16.1)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `doc`

(ANN401)